variables.tf

```diff
@@ -43,6 +45,24 @@ variable "project_id" {
 variable "dataset_labels" {
   description = "Key value pairs in a map for dataset labels"
   type        = map(string)
+  default     = null
+}
+
+# Format: list(objects)
+# domain: A domain to grant access to.
+# group_by_email: An email address of a Google Group to grant access to.
+# user_by_email:  An email address of a user to grant access to.
+# group_by_email: An email address of a Google Group to grant access to.
+# special_group: A special group to grant access to.
+variable "access" {
+  description = "An array of objects that define dataset access for one or more entities."
+  type        = any
+
+  # At least one owner access is required.
```

Comment:
Is this actually accurate? This will cause a diff in all existing plans which I'd like to avoid.

Reply:
No, it's not (hence my comment above of making a major release along with this PR).

The defaults of BQ are actually quite bad which is why I made this PR to begin with as we would like to use this module for our e2e solutions. The default BQ adds the calling user as an owner on the dataset which is a big no-no for us (access should only be granted via groups). This default is much more secure and in line with how other GCP resources.

I thought it would be ok to improve the default for CFT as better defaults was one of the goals. But if you prefer to avoid diff I can just set this to null as default.